### PR TITLE
Fix issue with second derivative values on faces.

### LINKED ISSUE #6327: Look into issue with second derivatives in IntegratedBCs
@jiangwen84 [reported](https://groups.google.com/d/msg/moose-users/_2Q6l6IloDE/4BMQbH-HDgAJ) an issue with accessing second derivatives from an IntegratedBC object.  

Off-list, we already confirmed that the second derivatives are being computed correctly in libmesh, but there is some issue with both the solution (accessed by `second()`) and test function (accessed by `secondTest()`) second derivative values in IntegratedBCs.

A possible hint to what is going wrong is that the values in `_var.secondPhiFace()` seem to be correct, i.e. the `MooseVariable` has computed the values correctly, so it's possible that it's just the Interface which is doing the wrong thing.  On the other hand, `_var.secondSln()` seems to have incorrect values, so it's not just a matter of accessing stuff on `MooseVariable` directly to fix the issue.

### PR DESCRIPTION
**Pull request by jwpeterson**
This PR makes it possible to get test and trial function second derivative values from `MooseVariableInterface` and fixes an issue where second derivatives were not being properly zeroed on faces in Steady simulations.  Also adds a test to ensure that this functionality keeps working.

Thanks to @jiangwen84 and @andrsd for their assistance and example code.

Closes #6327.

### CODE REVIEW

File: framework/src/base/MooseVariableInterface.C [hunk 1]
```diff
@@ -155,6 +155,16 @@ MooseVariableInterface::secondTest()
   return const_cast<VariableTestSecond &>(_variable->secondPhi());
 }
 
+VariableTestSecond &
+MooseVariableInterface::secondTestFace()
+{
+  if (_nodal)
+    mooseError("Nodal variables do not have second derivatives");
+
+  return const_cast<VariableTestSecond &>(_variable->secondPhiFace());
```
Comment (dschwen) [inline framework/src/base/MooseVariableInterface.C]: Why don't you make that method return a const reference? We should switch all of these methods over at some time.

Comment (jwpeterson) [inline framework/src/base/MooseVariableInterface.C]: I just assumed we were doing this for some reason.  I think I'd rather update them all in a separate PR anyway -- it's going to change a lot of user code when we do...

Comment (dschwen) [inline framework/src/base/MooseVariableInterface.C]: Yeah, I guess having a mix of const and non-const would be weird.

**Comment by jwpeterson** [comment]
Hm... I guess I can see why this might change the results of Laplacian jump indicator since it uses `_second_u_neighbor`, but not sure why that would only be uncovered by the threaded runs...

**Comment by jwpeterson** [comment]
Nevermind, nothing to do with threads... that's just the name of the target.  I think the `LaplacianJumpIndicator` just needs to be regolded.

**Comment by jwpeterson** [comment]
Yeah, so the error indicator really made no sense before: the exact solution to this test problem is nearly linear, so the Laplacian error jump should be very small for Hermite elements.  Instead the error looked like this:

before


But now looks like this (note the scale):

after

**Comment by jwpeterson** [comment]
Damn, really small difference in the error indicator as computed in serial vs. parallel now:
```
indicators/laplacian_jump_indicator.test: Element variables:
indicators/laplacian_jump_indicator.test:    error  rel diff:  6.4447937e-06 ~  6.4547313e-06 = 1.53958e-03 (block 0, elmt 45)
```

**Comment by permcody** [comment]
:+1:

**Pull request merged**